module/x/gravity/keeper/msg_server.go

```diff
@@ -19,13 +19,116 @@ type msgServer struct {
 	Keeper
 }
 
+var _ types.MsgServer = msgServer{}
+
 // NewMsgServerImpl returns an implementation of the gov MsgServer interface
 // for the provided Keeper.
 func NewMsgServerImpl(keeper Keeper) types.MsgServer {
 	return &msgServer{Keeper: keeper}
 }
 
-var _ types.MsgServer = msgServer{}
+func (k msgServer) SendToEVM(ctx context.Context, evm *types.MsgSendToEVM) (*types.MsgSendToEVMResponse, error) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (k msgServer) CancelSendToEVM(ctx context.Context, evm *types.MsgCancelSendToEVM) (*types.MsgCancelSendToEVMResponse, error) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (k msgServer) RequestEVMBatchTx(ctx context.Context, tx *types.MsgRequestEVMBatchTx) (*types.MsgRequestEVMBatchTxResponse, error) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (k msgServer) SubmitEVMTxConfirmation(ctx context.Context, confirmation *types.MsgSubmitEVMTxConfirmation) (*types.MsgSubmitEVMTxConfirmationResponse, error) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (k msgServer) SubmitEVMEvent(ctx context.Context, event *types.MsgSubmitEVMEvent) (*types.MsgSubmitEVMEventResponse, error) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (k msgServer) SetDelegateChainKeys(c context.Context, msg *types.MsgDelegateChainKeys) (*types.MsgDelegateChainKeysResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+
+	valAddr, err := sdk.ValAddressFromBech32(msg.ValidatorAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	orchAddr, err := sdk.AccAddressFromBech32(msg.OrchestratorAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	ethAddr := common.HexToAddress(msg.EvmAddress)
+
+	// ensure that the validator exists
+	if k.Keeper.StakingKeeper.Validator(ctx, valAddr) == nil {
+		return nil, sdkerrors.Wrap(stakingtypes.ErrNoValidatorFound, valAddr.String())
+	}
+
+	// check if the Ethereum address is currently not used
+	validators := k.getValidatorsByEthereumAddress(ctx, ethAddr)
```

Comment:
This and the remainder of the file are still calling the Ethereum keeper functions. If you want backwards compatibility I believe you'll need a separate keyspace.